clang/test/SemaSYCL/sycl-device.cpp

```diff
@@ -21,11 +21,17 @@ namespace {
 }
 
 class A {
-  __attribute__((sycl_device)) // expected-error {{'sycl_device' attribute cannot be applied to a class member function}}
+  __attribute__((sycl_device))
   A() {}
 
-  __attribute__((sycl_device)) // expected-error {{'sycl_device' attribute cannot be applied to a class member function}}
-  int func3() {}
+  __attribute__((sycl_device)) void func3() {}
+};
+
+class B {
+public:
```

Comment:
Can you add tests for what the behavior (I think from a CodeGen perspective) of what happens to:

override functions marked with these attributes (where the 'base' does NOT have them)
override functions marked with these attributes (where hte base DOES have them)
unmarked override functions (where the base DOES have them)
marked/unmarked 'final' functions in the above.

Reply:
Sure, added for both attributes.